api_server.py

```
@@ -150,19 +164,27 @@ def total_hrs(p):
-        rows = conn.execute(
-            f"SELECT role, skills FROM crew WHERE id IN ({placeholders})", available_ids
-        ).fetchall()
+
+    if crew_dicts:
+        # Use pre-fetched data to avoid redundant DB query
+        rows = [c for c in crew_dicts if c["id"] in available_ids]
+    else:
+        with get_db() as conn:
+            placeholders = ",".join("?" * len(available_ids))
+            rows = conn.execute(
+                f"SELECT role, skills FROM crew WHERE id IN ({placeholders})",
```

Ruff (0.14.14)

[error] Possible SQL injection vector through string-based query construction

(S608)